frontend/src/components/object/ObjectFileDetails.vue

```diff
@@ -42,12 +43,31 @@ const obj: Ref<COMSObject | undefined> = ref(undefined);
 const bucketId: Ref<string> = ref('');
 
 // Actions
+const router = useRouter();
+
 const showPermissions = async (objectId: string) => {
   permissionsVisible.value = true;
   permissionsObjectId.value = objectId;
   permissionsObjectName.value = metadataStore.findValue(objectId, 'name') || '';
 };
 
+onBeforeMount( async () => {
+  if( props.objectId ) {
+    const head = await objectStore.headObject(props.objectId);
+    let isPublic = head?.status === 204;
+
+    await permissionStore.fetchBucketPermissions({userId: getUserId.value});
+    await objectStore.fetchObjects({objectId: props.objectId, userId: getUserId.value});
+    const obj = objectStore.findObjectById(props.objectId);
+    const bucketId = obj?.bucketId;
+
+    if( !isPublic &&
+          ( !obj || !permissionStore.isObjectActionAllowed(obj.id, getUserId.value, Permissions.READ, bucketId) ) ) {
+      router.replace({ name: RouteNames.FORBIDDEN });
```

Comment:
This may need to be a router.push instead so that we have history of where we were prior to getting declined; replace deletes where we came from in the history.

Reply:
Leaving this as replace for now.